spec/models/payroll_run_spec.rb

```diff
@@ -43,16 +43,36 @@
   end
 
   describe ".create_with_claims!" do
-    let(:claims) { Policies.all.map { |policy| create(:claim, :approved, policy: policy) } }
+    let(:claims) do
+      Policies.all.map do |policy|
+        create(:claim, :approved, policy: policy, national_insurance_number: generate(:national_insurance_number))
+      end
+    end
+    subject!(:payroll_run) { PayrollRun.create_with_claims!(claims, created_by: user) }
 
     it "creates a payroll run with payments and populates the award_amount" do
-      payroll_run = PayrollRun.create_with_claims!(claims, created_by: user)
-
       expect(payroll_run.reload.created_by.id).to eq(user.id)
       expect(payroll_run.claims).to match_array(claims)
       expect(claims[0].payment.award_amount).to eq(claims[0].award_amount)
       expect(claims[1].payment.award_amount).to eq(claims[1].award_amount)
     end
+
+    context "with multiple claims from the same National Insurance number" do
+      let(:matching_claims) do
+        national_insurance_number = generate(:national_insurance_number)
+        [
+          create(:claim, :approved, policy: StudentLoans, national_insurance_number: national_insurance_number),
+          create(:claim, :approved, policy: MathsAndPhysics, national_insurance_number: national_insurance_number),
+        ]
+      end
+      let(:other_claim) { create(:claim, :approved) }
+      let(:claims) { matching_claims + [other_claim] }
+
+      it "groups them into a single payment" do
+        expect(matching_claims[0].reload.payment.claims).to match_array(matching_claims)
```

Comment:
I think this spec would be more confidence-inducing if it checked `payroll_run.payments`, that there were two, and that one was against the two matching claims and one was the other claim.

Reply:
:+1: